vyper/codegen/ir_node.py

```diff
@@ -464,6 +464,11 @@ def referenced_variables(self):
 
         return ret
 
+    @cached_property
+    def contains_risky_call(self):
```

Comment:
this is because it could call back?

Reply:
yea, bc of reentrancy we have to assume all variables are tainted

we could restrict it to only variables which are touchable from (reentrant) external functions, but that requires more analysis.

once venom optimizer gets more sophisticated we can clean this up by *always* writing to temp buffer and then letting memory alias analysis optimize out the intermediate writes if it deems they are safe